src/storage/buffer_manager.cpp

```diff
@@ -268,6 +268,12 @@ shared_ptr<BlockHandle> BlockManager::RegisterBlock(block_id_t block_id) {
 	return result;
 }
 
+shared_ptr<BlockHandle> BlockManager::RegisterMetaBlock(block_id_t block_id) {
+	auto handle = RegisterBlock(block_id);
+	meta_blocks[block_id] = handle;
```

Comment:
This cache will always persist - perhaps we should clear it after the metadata is read (by calling `meta_blocks.clear()` after the initial load of a database is completed)?